### PR TITLE
Remove useless abstraction around XmlReaderSettings

### PR DESCRIPTION
**Pull request by Bond-009**
This removes the amount of stuff that needs to be passed around
Also removes some unneeded `ManagedFileSystem` usage

### CODE REVIEW

File: MediaBrowser.Providers/People/TvdbPersonImageProvider.cs [hunk 1]
```diff
@@ -99,13 +93,15 @@ private RemoteImageInfo GetImageFromSeriesData(Series series, string personName,
 
         private RemoteImageInfo GetImageInfo(string xmlFile, string personName, CancellationToken cancellationToken)
         {
-            var settings = _xmlSettings.Create(false);
-
-            settings.CheckCharacters = false;
-            settings.IgnoreProcessingInstructions = true;
-            settings.IgnoreComments = true;
+            var settings = new XmlReaderSettings()
```
Comment (JustAMan) [inline MediaBrowser.Providers/People/TvdbPersonImageProvider.cs]: I see this code being copied over and over, can we have it in one place like a static function?..

File: MediaBrowser.Providers/TV/TheTVDB/TvdbPrescanTask.cs [hunk 2]
```diff
@@ -182,41 +179,41 @@ public async Task Run(IProgress<double> progress, CancellationToken cancellation
         /// <returns>System.String.</returns>
         private string GetUpdateTime(Stream response)
         {
-            var settings = _xmlSettings.Create(false);
-
-            settings.CheckCharacters = false;
-            settings.IgnoreProcessingInstructions = true;
-            settings.IgnoreComments = true;
+            // Use XmlReader for best performance
```
Comment (JustAMan) [inline MediaBrowser.Providers/TV/TheTVDB/TvdbPrescanTask.cs]: this comment seems useless

File: MediaBrowser.Providers/TV/TheTVDB/TvdbPrescanTask.cs [hunk 3]
```diff
@@ -248,65 +245,65 @@ private string GetUpdateTime(Stream response)
...
         {
             string updateTime = null;
             var idList = new List<string>();
 
-            var settings = _xmlSettings.Create(false);
-
-            settings.CheckCharacters = false;
-            settings.IgnoreProcessingInstructions = true;
-            settings.IgnoreComments = true;
+            // Use XmlReader for best performance
```
Comment (JustAMan) [inline MediaBrowser.Providers/TV/TheTVDB/TvdbPrescanTask.cs]: this too

File: MediaBrowser.Providers/TV/TheTVDB/TvdbSeasonImageProvider.cs [hunk 4]
```diff
@@ -88,62 +84,60 @@ public async Task<IEnumerable<RemoteImageInfo>> GetImages(BaseItem item, Cancell
...
 
-        internal static IEnumerable<RemoteImageInfo> GetImages(string xmlPath, string preferredLanguage, int seasonNumber, IXmlReaderSettingsFactory xmlReaderSettingsFactory, IFileSystem fileSystem, CancellationToken cancellationToken)
+        internal static IEnumerable<RemoteImageInfo> GetImages(string xmlPath, string preferredLanguage, int seasonNumber, CancellationToken cancellationToken)
         {
-            var settings = xmlReaderSettingsFactory.Create(false);
-
-            settings.CheckCharacters = false;
-            settings.IgnoreProcessingInstructions = true;
-            settings.IgnoreComments = true;
+            // Use XmlReader for best performance
```
Comment (JustAMan) [inline MediaBrowser.Providers/TV/TheTVDB/TvdbSeasonImageProvider.cs]: and this also seems useless

File: MediaBrowser.Providers/TV/TheTVDB/TvdbSeriesImageProvider.cs [hunk 5]
```diff
@@ -90,57 +87,55 @@ public async Task<IEnumerable<RemoteImageInfo>> GetImages(BaseItem item, Cancell
 
         private IEnumerable<RemoteImageInfo> GetImages(string xmlPath, string preferredLanguage, CancellationToken cancellationToken)
         {
-            var settings = _xmlReaderSettingsFactory.Create(false);
-
-            settings.CheckCharacters = false;
-            settings.IgnoreProcessingInstructions = true;
-            settings.IgnoreComments = true;
+            // Use XmlReader for best performance
```
Comment (JustAMan) [inline MediaBrowser.Providers/TV/TheTVDB/TvdbSeriesImageProvider.cs]: again useless

File: MediaBrowser.Providers/TV/TheTVDB/TvdbSeriesProvider.cs [hunk 6]
```diff
@@ -586,58 +583,58 @@ private async Task<List<RemoteSearchResult>> FindSeriesInternal(string name, str
             {
                 using (var stream = response.Content)
                 {
-                    var settings = _xmlSettings.Create(false);
-
-                    settings.CheckCharacters = false;
-                    settings.IgnoreProcessingInstructions = true;
-                    settings.IgnoreComments = true;
+                    // Use XmlReader for best performance
```
Comment (JustAMan) [inline MediaBrowser.Providers/TV/TheTVDB/TvdbSeriesProvider.cs]: it's starting to get a bit tiring to see this comment all over the place

File: MediaBrowser.Providers/TV/TheTVDB/TvdbSeriesProvider.cs [hunk 7]
```diff
@@ -819,83 +816,81 @@ private string GetComparableName(string name)
 
         private void FetchSeriesInfo(MetadataResult<Series> result, string seriesXmlPath, CancellationToken cancellationToken)
         {
-            var settings = _xmlSettings.Create(false);
-
-            settings.CheckCharacters = false;
-            settings.IgnoreProcessingInstructions = true;
-            settings.IgnoreComments = true;
+            // Use XmlReader for best performance
```
Comment (JustAMan) [inline MediaBrowser.Providers/TV/TheTVDB/TvdbSeriesProvider.cs]: remove it?

File: MediaBrowser.Providers/TV/TheTVDB/TvdbSeriesProvider.cs [hunk 8]
```diff
@@ -973,53 +968,51 @@ private void FetchSeriesInfo(MetadataResult<Series> result, string seriesXmlPath
         /// <param name="actorsXmlPath">The actors XML path.</param>
         private void FetchActors(MetadataResult<Series> result, string actorsXmlPath)
         {
-            var settings = _xmlSettings.Create(false);
-
-            settings.CheckCharacters = false;
-            settings.IgnoreProcessingInstructions = true;
-            settings.IgnoreComments = true;
+            // Use XmlReader for best performance
```
Comment (JustAMan) [inline MediaBrowser.Providers/TV/TheTVDB/TvdbSeriesProvider.cs]: remove it?

File: MediaBrowser.Providers/TV/TheTVDB/TvdbSeriesProvider.cs [hunk 9]
```diff
@@ -1349,59 +1342,60 @@ private void FetchDataFromSeriesNode(MetadataResult<Series> result, XmlReader re
         /// <returns>Task.</returns>
         private async Task ExtractEpisodes(string seriesDataPath, string xmlFile, long? lastTvDbUpdateTime)
         {
-            var settings = _xmlSettings.Create(false);
-
-            settings.CheckCharacters = false;
-            settings.IgnoreProcessingInstructions = true;
-            settings.IgnoreComments = true;
+            // Use XmlReader for best performance
```
Comment (JustAMan) [inline MediaBrowser.Providers/TV/TheTVDB/TvdbSeriesProvider.cs]: remove it?

File: MediaBrowser.Providers/TV/TheTVDB/TvdbSeriesProvider.cs [hunk 10]
```diff
@@ -1349,59 +1342,60 @@ private void FetchDataFromSeriesNode(MetadataResult<Series> result, XmlReader re
...
         }
 
         private async Task SaveEpsiodeXml(string seriesDataPath, string xml, long? lastTvDbUpdateTime)
         {
-            var settings = _xmlSettings.Create(false);
-
-            settings.CheckCharacters = false;
-            settings.IgnoreProcessingInstructions = true;
-            settings.IgnoreComments = true;
+            // Use XmlReader for best performance
```
Comment (JustAMan) [inline MediaBrowser.Providers/TV/TheTVDB/TvdbSeriesProvider.cs]: remove it?

File: MediaBrowser.XbmcMetadata/Providers/AlbumNfoProvider.cs [hunk 11]
```diff
@@ -15,20 +14,18 @@ public class AlbumNfoProvider : BaseNfoProvider<MusicAlbum>
         private readonly ILogger _logger;
         private readonly IConfigurationManager _config;
         private readonly IProviderManager _providerManager;
-        protected IXmlReaderSettingsFactory XmlReaderSettingsFactory { get; private set; }
 
-        public AlbumNfoProvider(IFileSystem fileSystem, ILogger logger, IConfigurationManager config, IProviderManager providerManager, IXmlReaderSettingsFactory xmlReaderSettingsFactory)
+        public AlbumNfoProvider(IFileSystem fileSystem, ILogger logger, IConfigurationManager config, IProviderManager providerManager)
```
Comment (JustAMan) [inline MediaBrowser.XbmcMetadata/Providers/AlbumNfoProvider.cs]: drop `fileSystem`?

File: MediaBrowser.XbmcMetadata/Providers/ArtistNfoProvider.cs [hunk 12]
```diff
@@ -15,20 +14,18 @@ public class ArtistNfoProvider : BaseNfoProvider<MusicArtist>
         private readonly ILogger _logger;
         private readonly IConfigurationManager _config;
         private readonly IProviderManager _providerManager;
-        protected IXmlReaderSettingsFactory XmlReaderSettingsFactory { get; private set; }
 
-        public ArtistNfoProvider(IFileSystem fileSystem, ILogger logger, IConfigurationManager config, IProviderManager providerManager, IXmlReaderSettingsFactory xmlReaderSettingsFactory)
+        public ArtistNfoProvider(IFileSystem fileSystem, ILogger logger, IConfigurationManager config, IProviderManager providerManager)
```
Comment (JustAMan) [inline MediaBrowser.XbmcMetadata/Providers/ArtistNfoProvider.cs]: drop `fileSystem`?

File: MediaBrowser.XbmcMetadata/Providers/BaseVideoNfoProvider.cs [hunk 13]
```diff
@@ -17,15 +16,13 @@ public class BaseVideoNfoProvider<T> : BaseNfoProvider<T>
         private readonly ILogger _logger;
         private readonly IConfigurationManager _config;
         private readonly IProviderManager _providerManager;
-        protected IXmlReaderSettingsFactory XmlReaderSettingsFactory { get; private set; }
 
-        public BaseVideoNfoProvider(IFileSystem fileSystem, ILogger logger, IConfigurationManager config, IProviderManager providerManager, IXmlReaderSettingsFactory xmlReaderSettingsFactory)
+        public BaseVideoNfoProvider(IFileSystem fileSystem, ILogger logger, IConfigurationManager config, IProviderManager providerManager)
```
Comment (JustAMan) [inline MediaBrowser.XbmcMetadata/Providers/BaseVideoNfoProvider.cs]: drop `fileSystem`?

File: MediaBrowser.XbmcMetadata/Providers/EpisodeNfoProvider.cs [hunk 14]
```diff
@@ -16,22 +15,20 @@ public class EpisodeNfoProvider : BaseNfoProvider<Episode>
         private readonly ILogger _logger;
         private readonly IConfigurationManager _config;
         private readonly IProviderManager _providerManager;
-        protected IXmlReaderSettingsFactory XmlReaderSettingsFactory { get; private set; }
 
-        public EpisodeNfoProvider(IFileSystem fileSystem, ILogger logger, IConfigurationManager config, IProviderManager providerManager, IXmlReaderSettingsFactory xmlReaderSettingsFactory)
+        public EpisodeNfoProvider(IFileSystem fileSystem, ILogger logger, IConfigurationManager config, IProviderManager providerManager)
```
Comment (JustAMan) [inline MediaBrowser.XbmcMetadata/Providers/EpisodeNfoProvider.cs]: drop `fileSystem`?

File: MediaBrowser.XbmcMetadata/Providers/MovieNfoProvider.cs [hunk 15]
```diff
@@ -3,28 +3,30 @@
...
 using MediaBrowser.Model.IO;
-using MediaBrowser.Model.Xml;
 using Microsoft.Extensions.Logging;
 
 namespace MediaBrowser.XbmcMetadata.Providers
 {
     public class MovieNfoProvider : BaseVideoNfoProvider<Movie>
     {
-        public MovieNfoProvider(IFileSystem fileSystem, ILogger logger, IConfigurationManager config, IProviderManager providerManager, IXmlReaderSettingsFactory xmlReaderSettingsFactory) : base(fileSystem, logger, config, providerManager, xmlReaderSettingsFactory)
+        public MovieNfoProvider(IFileSystem fileSystem, ILogger logger, IConfigurationManager config, IProviderManager providerManager)
```
Comment (JustAMan) [inline MediaBrowser.XbmcMetadata/Providers/MovieNfoProvider.cs]: drop `fileSystem`?

File: MediaBrowser.XbmcMetadata/Providers/MovieNfoProvider.cs [hunk 16]
```diff
@@ -3,28 +3,30 @@
...
+            : base(fileSystem, logger, config, providerManager)
         {
         }
     }
 
     public class MusicVideoNfoProvider : BaseVideoNfoProvider<MusicVideo>
     {
-        public MusicVideoNfoProvider(IFileSystem fileSystem, ILogger logger, IConfigurationManager config, IProviderManager providerManager, IXmlReaderSettingsFactory xmlReaderSettingsFactory) : base(fileSystem, logger, config, providerManager, xmlReaderSettingsFactory)
+        public MusicVideoNfoProvider(IFileSystem fileSystem, ILogger logger, IConfigurationManager config, IProviderManager providerManager)
```
Comment (JustAMan) [inline MediaBrowser.XbmcMetadata/Providers/MovieNfoProvider.cs]: drop `fileSystem`?

File: MediaBrowser.XbmcMetadata/Providers/MovieNfoProvider.cs [hunk 17]
```diff
@@ -3,28 +3,30 @@
...
+            : base(fileSystem, logger, config, providerManager)
         {
         }
     }
 
     public class VideoNfoProvider : BaseVideoNfoProvider<Video>
     {
-        public VideoNfoProvider(IFileSystem fileSystem, ILogger logger, IConfigurationManager config, IProviderManager providerManager, IXmlReaderSettingsFactory xmlReaderSettingsFactory) : base(fileSystem, logger, config, providerManager, xmlReaderSettingsFactory)
+        public VideoNfoProvider(IFileSystem fileSystem, ILogger logger, IConfigurationManager config, IProviderManager providerManager)
```
Comment (JustAMan) [inline MediaBrowser.XbmcMetadata/Providers/MovieNfoProvider.cs]: drop `fileSystem`?

File: MediaBrowser.XbmcMetadata/Providers/SeasonNfoProvider.cs [hunk 18]
```diff
@@ -15,26 +14,22 @@ public class SeasonNfoProvider : BaseNfoProvider<Season>
         private readonly ILogger _logger;
         private readonly IConfigurationManager _config;
         private readonly IProviderManager _providerManager;
-        protected IXmlReaderSettingsFactory XmlReaderSettingsFactory { get; private set; }
 
-        public SeasonNfoProvider(IFileSystem fileSystem, ILogger logger, IConfigurationManager config, IProviderManager providerManager, IXmlReaderSettingsFactory xmlReaderSettingsFactory)
+        public SeasonNfoProvider(IFileSystem fileSystem, ILogger logger, IConfigurationManager config, IProviderManager providerManager)
```
Comment (JustAMan) [inline MediaBrowser.XbmcMetadata/Providers/SeasonNfoProvider.cs]: drop `fileSystem`?

File: MediaBrowser.XbmcMetadata/Providers/SeriesNfoProvider.cs [hunk 19]
```diff
@@ -15,27 +14,21 @@ public class SeriesNfoProvider : BaseNfoProvider<Series>
         private readonly ILogger _logger;
         private readonly IConfigurationManager _config;
         private readonly IProviderManager _providerManager;
-        private readonly IFileSystem _fileSystem;
-        protected IXmlReaderSettingsFactory XmlReaderSettingsFactory { get; private set; }
 
-        public SeriesNfoProvider(IFileSystem fileSystem, ILogger logger, IConfigurationManager config, IProviderManager providerManager, IFileSystem fileSystem1, IXmlReaderSettingsFactory xmlReaderSettingsFactory)
+        public SeriesNfoProvider(IFileSystem fileSystem, ILogger logger, IConfigurationManager config, IProviderManager providerManager)
```
Comment (JustAMan) [inline MediaBrowser.XbmcMetadata/Providers/SeriesNfoProvider.cs]: drop `fileSystem`?

File: MediaBrowser.XbmcMetadata/Savers/AlbumNfoSaver.cs [hunk 20]
```diff
@@ -94,7 +93,8 @@ protected override List<string> GetTagsUsed(BaseItem item)
             return list;
         }
 
-        public AlbumNfoSaver(IFileSystem fileSystem, IServerConfigurationManager configurationManager, ILibraryManager libraryManager, IUserManager userManager, IUserDataManager userDataManager, ILogger logger, IXmlReaderSettingsFactory xmlReaderSettingsFactory) : base(fileSystem, configurationManager, libraryManager, userManager, userDataManager, logger, xmlReaderSettingsFactory)
+        public AlbumNfoSaver(IFileSystem fileSystem, IServerConfigurationManager configurationManager, ILibraryManager libraryManager, IUserManager userManager, IUserDataManager userDataManager, ILogger logger)
```
Comment (JustAMan) [inline MediaBrowser.XbmcMetadata/Savers/AlbumNfoSaver.cs]: drop `fileSystem`?

File: MediaBrowser.XbmcMetadata/Savers/ArtistNfoSaver.cs [hunk 21]
```diff
@@ -85,7 +84,8 @@ protected override List<string> GetTagsUsed(BaseItem item)
             return list;
         }
 
-        public ArtistNfoSaver(IFileSystem fileSystem, IServerConfigurationManager configurationManager, ILibraryManager libraryManager, IUserManager userManager, IUserDataManager userDataManager, ILogger logger, IXmlReaderSettingsFactory xmlReaderSettingsFactory) : base(fileSystem, configurationManager, libraryManager, userManager, userDataManager, logger, xmlReaderSettingsFactory)
+        public ArtistNfoSaver(IFileSystem fileSystem, IServerConfigurationManager configurationManager, ILibraryManager libraryManager, IUserManager userManager, IUserDataManager userDataManager, ILogger logger)
```
Comment (JustAMan) [inline MediaBrowser.XbmcMetadata/Savers/ArtistNfoSaver.cs]: drop `fileSystem`?

File: MediaBrowser.XbmcMetadata/Savers/BaseNfoSaver.cs [hunk 22]
```diff
@@ -97,10 +96,15 @@ public abstract class BaseNfoSaver : IMetadataFileSaver
 
         }.ToDictionary(i => i, StringComparer.OrdinalIgnoreCase);
 
-        protected BaseNfoSaver(IFileSystem fileSystem, IServerConfigurationManager configurationManager, ILibraryManager libraryManager, IUserManager userManager, IUserDataManager userDataManager, ILogger logger, IXmlReaderSettingsFactory xmlReaderSettingsFactory)
+        protected BaseNfoSaver(
+            IFileSystem fileSystem,
```
Comment (JustAMan) [inline MediaBrowser.XbmcMetadata/Savers/BaseNfoSaver.cs]: drop `fileSystem`?

File: MediaBrowser.XbmcMetadata/Savers/BaseNfoSaver.cs [hunk 23]
```diff
@@ -961,52 +964,50 @@ private bool IsPersonType(PersonInfo person, string type)
 
         private void AddCustomTags(string path, List<string> xmlTagsUsed, XmlWriter writer, ILogger logger, IFileSystem fileSystem)
         {
-            var settings = XmlReaderSettingsFactory.Create(false);
-
-            settings.CheckCharacters = false;
-            settings.IgnoreProcessingInstructions = true;
-            settings.IgnoreComments = true;
+            // Use XmlReader for best performance
```
Comment (JustAMan) [inline MediaBrowser.XbmcMetadata/Savers/BaseNfoSaver.cs]: remove it?

File: MediaBrowser.XbmcMetadata/Savers/EpisodeNfoSaver.cs [hunk 24]
```diff
@@ -109,7 +108,8 @@ protected override List<string> GetTagsUsed(BaseItem item)
             return list;
         }
 
-        public EpisodeNfoSaver(IFileSystem fileSystem, IServerConfigurationManager configurationManager, ILibraryManager libraryManager, IUserManager userManager, IUserDataManager userDataManager, ILogger logger, IXmlReaderSettingsFactory xmlReaderSettingsFactory) : base(fileSystem, configurationManager, libraryManager, userManager, userDataManager, logger, xmlReaderSettingsFactory)
+        public EpisodeNfoSaver(IFileSystem fileSystem, IServerConfigurationManager configurationManager, ILibraryManager libraryManager, IUserManager userManager, IUserDataManager userDataManager, ILogger logger)
```
Comment (JustAMan) [inline MediaBrowser.XbmcMetadata/Savers/EpisodeNfoSaver.cs]: drop `fileSystem`?

File: MediaBrowser.XbmcMetadata/Savers/MovieNfoSaver.cs [hunk 25]
```diff
@@ -129,7 +128,8 @@ protected override List<string> GetTagsUsed(BaseItem item)
             return list;
         }
 
-        public MovieNfoSaver(IFileSystem fileSystem, IServerConfigurationManager configurationManager, ILibraryManager libraryManager, IUserManager userManager, IUserDataManager userDataManager, ILogger logger, IXmlReaderSettingsFactory xmlReaderSettingsFactory) : base(fileSystem, configurationManager, libraryManager, userManager, userDataManager, logger, xmlReaderSettingsFactory)
+        public MovieNfoSaver(IFileSystem fileSystem, IServerConfigurationManager configurationManager, ILibraryManager libraryManager, IUserManager userManager, IUserDataManager userDataManager, ILogger logger)
```
Comment (JustAMan) [inline MediaBrowser.XbmcMetadata/Savers/MovieNfoSaver.cs]: drop `fileSystem`?

File: MediaBrowser.XbmcMetadata/Savers/SeasonNfoSaver.cs [hunk 26]
```diff
@@ -66,9 +65,8 @@ protected override List<string> GetTagsUsed(BaseItem item)
             ILibraryManager libraryManager,
             IUserManager userManager,
             IUserDataManager userDataManager,
-            ILogger logger,
-            IXmlReaderSettingsFactory xmlReaderSettingsFactory)
-            : base(fileSystem, configurationManager, libraryManager, userManager, userDataManager, logger, xmlReaderSettingsFactory)
+            ILogger logger)
+            : base(fileSystem, configurationManager, libraryManager, userManager, userDataManager, logger)
```
Comment (JustAMan) [inline MediaBrowser.XbmcMetadata/Savers/SeasonNfoSaver.cs]: drop `fileSystem`?

File: MediaBrowser.XbmcMetadata/Savers/SeriesNfoSaver.cs [hunk 27]
```diff
@@ -83,7 +82,8 @@ protected override List<string> GetTagsUsed(BaseItem item)
             return list;
         }
 
-        public SeriesNfoSaver(IFileSystem fileSystem, IServerConfigurationManager configurationManager, ILibraryManager libraryManager, IUserManager userManager, IUserDataManager userDataManager, ILogger logger, IXmlReaderSettingsFactory xmlReaderSettingsFactory) : base(fileSystem, configurationManager, libraryManager, userManager, userDataManager, logger, xmlReaderSettingsFactory)
+        public SeriesNfoSaver(IFileSystem fileSystem, IServerConfigurationManager configurationManager, ILibraryManager libraryManager, IUserManager userManager, IUserDataManager userDataManager, ILogger logger)
```
Comment (JustAMan) [inline MediaBrowser.XbmcMetadata/Savers/SeriesNfoSaver.cs]: drop `fileSystem`?

File: MediaBrowser.XbmcMetadata/Savers/EpisodeNfoSaver.cs (hunk 24 above)
Comment (Bond-009) [reply MediaBrowser.XbmcMetadata/Savers/EpisodeNfoSaver.cs]: Can't the base class needs it atm.

File: MediaBrowser.XbmcMetadata/Savers/MovieNfoSaver.cs (hunk 25 above)
Comment (Bond-009) [reply MediaBrowser.XbmcMetadata/Savers/MovieNfoSaver.cs]: Same

File: MediaBrowser.XbmcMetadata/Savers/SeasonNfoSaver.cs (hunk 26 above)
Comment (Bond-009) [reply MediaBrowser.XbmcMetadata/Savers/SeasonNfoSaver.cs]: Same

File: MediaBrowser.XbmcMetadata/Savers/SeriesNfoSaver.cs (hunk 27 above)
Comment (Bond-009) [reply MediaBrowser.XbmcMetadata/Savers/SeriesNfoSaver.cs]: Same

File: MediaBrowser.XbmcMetadata/Parsers/MovieNfoParser.cs [hunk 28]
```diff
@@ -122,18 +122,9 @@ private void ParseSetXml(string xml, Movie movie)
...
-                    CheckCharacters = false,
-                    IgnoreProcessingInstructions = true,
-                    IgnoreComments = true
-                };
-
                 try
                 {
-                    // Use XmlReader for best performance
-                    using (var reader = XmlReader.Create(stringReader, settings))
+                    using (var reader = XmlReader.Create(stringReader, GetXmlReaderSettings()))
```
Comment (JustAMan) [inline MediaBrowser.XbmcMetadata/Parsers/MovieNfoParser.cs]: Why is `GetXmlReaderSettings()` used in only one or two places? I think it should be used everywhere settings are created...

Comment (JustAMan) [reply MediaBrowser.XbmcMetadata/Parsers/MovieNfoParser.cs]: You're still making that settings object everywhere else... maybe fix it for now anyway?

File: MediaBrowser.XbmcMetadata/Savers/EpisodeNfoSaver.cs (hunk 24 above)
Comment (JustAMan) [reply MediaBrowser.XbmcMetadata/Savers/EpisodeNfoSaver.cs]: Cannot it be dropped from base class as well?

**Comment by joshuaboniface** [comment]
@Bond-009 @JustAMan Where are we at with this one, proceed or close?

**Comment by JustAMan** [comment]
@Bond-009 you have a conflict here, otherwise it seems we're good to merge it.

**Comment by ploughpuff** [comment]
Guys, this push to master does not compile!

Can we look to fix the CI checks to trap issues like these please?

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0234	The type or namespace name 'Xml' does not exist in the namespace 'MediaBrowser.Model' (are you missing an assembly reference?)	MediaBrowser.Providers	C:\git_repos\jellyfin\MediaBrowser.Providers\Music\MusicBrainzAlbumProvider.cs	18	N/A

```

**Comment by anthonylavado** [comment]
@ploughpuff I think this was literally just fixed: https://github.com/jellyfin/jellyfin/pull/1098

**Pull request merged**